src/math/polysat/op_constraint.cpp

```diff
@@ -275,7 +275,7 @@ namespace polysat {
         }
         else {
             // forward propagation
-            SASSERT(!(pv.is_val() && qv.is_val() && rv.is_val()));
```

Comment:
disabling this code is independent of the fi functionality?

Reply:
Jep - In our benchmarks it performs poorly (the justification is weak)